pkg/ir/ir.go

```diff
@@ -0,0 +1,66 @@
+package ir
+
+import (
+	route "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
+)
+
+// Xds holds the intermediate representation of a Gateway and is
+// used by the xDS Translator to convert it into xDS resources
+type Xds struct {
```

Comment:
I think we should try to keep xDS-isms out of the IR as much as possible, so this should probably have a different name.

We could just call it `Gateway`,  `GatewayConfig` or something that doesn't overload a name, maybe?

Reply:
I envisioned this IR to be specific to xDS (consumed by the xDS translator that generates xDS), and another one for the Provisioner, if we want to club the two into one, agree, we need to come up with a higher level noun